x-pack/plugins/security_solution/public/detection_engine/rule_monitoring/components/execution_events_table/event_message_filter/event_message_filter.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ChangeEvent } from 'react';
+import React, { useCallback } from 'react';
+import { EuiFieldSearch } from '@elastic/eui';
+import * as i18n from './translations';
+
+interface EventMessageFilterProps {
+  value: string;
+  onChange: (value: string) => void;
+}
+
+export function EventMessageFilter({ value, onChange }: EventMessageFilterProps): JSX.Element {
```

Comment:
Also this component. Please use `React.memo`

Reply:
My answer the same as [here](https://github.com/elastic/kibana/pull/166070#discussion_r1324519971). Wrapping every possible component in `React.memo` doesn't guarantee a performance boost and it doesn't come for free as it also leads to increased memory consumption. Ideally we should wrap only heavy components where render function performs a lot of operations and it's usually true for data grids, lists, components higher in a component tree hierarchy.

Do you have some examples of troubles with such components?

Reply:
Agree with your point on using `memo` correctly. It does add more JS for the browser to parse and execute. Is there a reason you want to create a wrapper component around `EuiFieldSearch` instead of using it as is in `execution_events_table.tsx`

Reply:
`EventMessageFilter` solves an interfacing issue. It happens because components may return change events like `EuiFieldSearch`'s `onChange` accepts `ChangeEvent`. It's an implementation detail and should be incapsulated. This way `useFilter` only operates on actual values while implementation detail can rely on whatever component.  `handleChange` inside `EventMessageFilter` is an adapter transforming `ChangeEvent<HTMLInputElement>` into a simpler text value. This way we can change the implementation e.g. replace `EuiFieldSearch` with something else without changing anything outside of this component. Yes, `handleChange` could be defined in `ExecutionEventsTable` but it won't conform to `EventMessageFilter` and `LogLevelFilter` and break a consisting approach. It's not a case for `EuiSuperDatePicker` so I made according changes by using it directly. If it required some transformation it'd be logical to wrap it in a component.

Reply:
Thanks for the explanation @maximpn.